src/main/scala/sparkz/core/api/http/PeersApiRoute.scala

```diff
@@ -80,12 +117,47 @@ case class PeersApiRoute(peerManager: ActorRef,
     }
   }
 
-  def blacklistedPeers: Route = (path("blacklisted") & get) {
-    val result = askActor[Seq[InetAddress]](peerManager, GetBlacklistedPeers)
-      .map(x => BlacklistedPeers(x.map(_.toString)).asJson)
-    ApiResponse(result)
+  def addToBlacklist: Route = (path("blacklist") & post & withAuth & entity(as[Json])) { json =>
+    val maybeAddress = json.asString.flatMap(addressAndPortRegexp.findFirstMatchIn)
+
+    maybeAddress match {
+      case None => ApiError(StatusCodes.BadRequest, s"address $maybeAddress is not well formatted")
+
+      case Some(addressAndPort) =>
+        val host = InetAddress.getByName(addressAndPort.group(1))
+        val port = addressAndPort.group(2).toInt
+        peerManager ! AddToBlacklist(new InetSocketAddress(host, port))
+        ApiResponse.OK
+    }
+  }
+
+  def removePeer: Route = (path("peer") & delete & withAuth & entity(as[Json])) { json =>
+    val maybeAddress = json.asString.flatMap(addressAndPortRegexp.findFirstMatchIn)
+
+    maybeAddress match {
+      case None => ApiError(StatusCodes.BadRequest, s"address $maybeAddress is not well formatted")
+
+      case Some(addressAndPort) =>
+        val host = InetAddress.getByName(addressAndPort.group(1))
+        val port = addressAndPort.group(2).toInt
+        peerManager ! RemovePeer(new InetSocketAddress(host, port))
```

Comment:
do we need to disconnect from peer in case it is connected right now?
If not, can it lead to any database inconsistency, for example, when the removed but connected peers want to update its record?

Reply:
Good catch, I'll add the disconnect message